source/utilities.tex

```diff
@@ -15581,6 +15644,215 @@
+    range_formatter<@\exposid{element-type}@, charT> @\exposid{underlying_}@;                         // \expos
+  public:
+    constexpr formatter();
+
+    template<class ParseContext>
+      constexpr typename ParseContext::iterator
+        parse(ParseContext& ctx);
+
+    template<class FormatContext>
+      typename FormatContext::iterator
+        format(@\exposid{maybe-const-map}@& r, FormatContext& ctx) const;
+  };
+}
+\end{codeblock}
+
+\indexlibrarymember{formatter}{\exposid{range-default-formatter}}%
+\begin{itemdecl}
+constexpr formatter();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+Either:
+\begin{itemize}
+\item
+\exposid{element-type} is a specialization of \tcode{pair}, or
+\item
+\exposid{element-type} is a specialization of \tcode{tuple} and
+\tcode{tuple_size_v<T> == 2}.
+\end{itemize}
+
+\pnum
+\effects
+Equivalent to:
+\begin{codeblock}
+@\exposid{underlying_}@.set_brackets(@\exposid{STATICALLY-WIDEN}@<charT>("{"), @\exposid{STATICALLY-WIDEN}@<charT>("}"));
+@\exposid{underlying_}@.underlying().set_brackets({}, {});
+@\exposid{underlying_}@.underlying().set_separator(@\exposid{STATICALLY-WIDEN}@<charT>(": "));
+\end{codeblock}
+\end{itemdescr}
+
+\indexlibrarymember{parse}{\exposid{range-default-formatter}}%
+\begin{itemdecl}
+template<class ParseContext>
+  constexpr typename ParseContext::iterator
+    parse(ParseContext& ctx);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return \exposid{underlying_}.parse(ctx);}
+\end{itemdescr}
+
+\indexlibrarymember{format}{\exposid{range-default-formatter}}%
+\begin{itemdecl}
+template<class FormatContext>
+  typename FormatContext::iterator
+    format(@\exposid{maybe-const-map}@& r, FormatContext& ctx) const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return \exposid{underlying_}.format(r, ctx);}
+\end{itemdescr}
+
+\indexlibraryglobal{\exposid{range-default-formatter}}%
+\begin{codeblock}
+namespace std {
+  template<ranges::@\libconcept{input_range}@ R, class charT>
+  struct @\exposid{range-default-formatter}@<range_format::set, R, charT> {
+  private:
+    using @\exposid{maybe-const-set}@ = @\exposid{fmt-maybe-const}@<R, charT>;                          // \expos
+    range_formatter<remove_cvref_t<ranges::range_reference_t<maybe-const-set>>,
+                    charT> @\exposid{underlying_}@;                                         // \expos
+
+  public:
+    constexpr formatter();
+
+    template<class ParseContext>
+      constexpr typename ParseContext::iterator
+        parse(ParseContext& ctx);
+
+    template<class FormatContext>
+      typename FormatContext::iterator
+        format(@\exposid{maybe-const-set}@& r, FormatContext& ctx) const;
+  };
+}
+\end{codeblock}
+
+%% FIXME: what's the return type? No longer a ctor.
+\indexlibrarymember{formatter}{\exposid{range-default-formatter}}%
+\begin{itemdecl}
+constexpr formatter();
```

Comment:
```suggestion
\indexlibrarymember{formatter}{\exposid{range-default-formatter}}%
\begin{itemdecl}
constexpr @\exposid{range-default-formatter}@();
```
Good catch. I hadn't noticed. It'd have been easy to catch if the edits and move weren't mingled in the same commit. The editor should be able to deal with those commits just like with fixup commits.

Reply:
Yeah, obviously the answer to the question is "there's no return type, it's still supposed to be a ctor". The intention was not to change it from a ctor to a normal member function with a missing return type. The ctor should just have been renamed along with its enclosing class, but LWG missed that.

Reply:
Thanks guys, fixed